.github/workflows/ValidatePullRequest.yml

```diff
@@ -5,30 +5,59 @@ name: Validate Pull Request
 on:
   pull_request:
     branches: [main, "release/**"]
-    paths-ignore:
-      - '**.md'
-      - '**.txt'
   merge_group:
 
 permissions:
   id-token: write
   contents: read
 
 jobs:
+  docs-pr:
+    runs-on: ubuntu-latest
+    outputs:
+      docs-only: ${{ steps.docs-only.outputs.result }}
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            docs:
+              - '**/*.md'
+              - '**/*.txt'
+            all:
+              - '**/*'
+      - uses: actions/github-script@v7
+        id: docs-only
+        with:
+          script: |
+            let docs_file_count = ${{steps.changes.outputs.docs_count}};
+            let all_file_count = ${{steps.changes.outputs.all_count}};
+            return all_file_count === docs_file_count;
+          result-encoding: string
 
   rust:
+    needs:
+      - docs-pr
     uses: ./.github/workflows/dep_rust.yml
     secrets: inherit
+    with: 
+      docs_only: ${{needs.docs-pr.outputs.docs-only != 'true'}}
```

Comment:
All these `'true'` need to change to `'false'`, and this PR should be running all the workflows as it's not a doc only change. Currently it's skipping all workflows.

Reply:
Also, I don't know if this should be a string comparison or a boolean :-\

Reply:
string comparison is the answer

Reply:
I think I had it reversed to test it and then started chasing my tail.